tests/unit/flow-construct/test_flow.py

```diff
@@ -827,3 +827,15 @@ def test_connect_to_predecessor():
     assert len(f._pod_nodes['gateway'].head_args.hosts_in_connect) == 0
     assert len(f._pod_nodes['pod1'].head_args.hosts_in_connect) == 0
     assert len(f._pod_nodes['pod2'].head_args.hosts_in_connect) == 1
+
+
+def test_flow_grpc_with_shard():
+    with pytest.raises(
+        NotImplementedError, match='GRPC data runtime does not support sharding'
+    ):
+        Flow(grpc_data_requests=True).add(shards=2)
+        Flow(grpc_data_requests=True).add(shards=None)
+
+    Flow(grpc_data_requests=False).add(shards=1)
+    Flow(grpc_data_requests=False).add()
+    Flow(grpc_data_requests=False, infrastructure=InfrastructureType.K8S).add(shards=2)
```

Comment:
```suggestion
    Flow(grpc_data_requests=True, infrastructure=InfrastructureType.K8S).add(shards=2)
```

The point is to validate check grpc and K8s is a legal combination

Reply:
Done.